File: frontend.v3/src/app/main/models/app-sync-data.ts
import { BookData } from '../../modules/book/models/book-data';
import { CollectionData } from '../../modules/collection/models/collection-data';
import { SyncData } from './sync-data';

export interface AppSyncData {
  books: SyncData<BookData>;
  collections: SyncData<CollectionData>;
}

File: backend.v2/backend.v2/Models/AppSyncData.cs
namespace backend.v2.Models
{
    /// <summary>
    /// Модель для синхронизации устройства.
    /// </summary>
    public class AppSyncData
    {
        /// <summary>
        /// Данные для синхронизации книг.
        /// </summary>
        public SyncData<Book> Books {get; set;}
        
        /// <summary>
        /// Данные для синхронизации коллекций.
        /// </summary>
        public SyncData<Collection> Collections {get; set;}
    }
}